django/db/models/query.py

```diff
@@ -933,26 +933,27 @@ def bulk_update(self, objs, fields, batch_size=None):
         batch_size = min(batch_size, max_batch_size) if batch_size else max_batch_size
         requires_casting = connection.features.requires_casted_case_in_updates
         batches = (objs[i : i + batch_size] for i in range(0, len(objs), batch_size))
-        updates = []
-        for batch_objs in batches:
-            update_kwargs = {}
-            for field in fields:
-                when_statements = []
-                for obj in batch_objs:
-                    attr = getattr(obj, field.attname)
-                    if not hasattr(attr, "resolve_expression"):
-                        attr = Value(attr, output_field=field)
-                    when_statements.append(When(pk=obj.pk, then=attr))
-                case_statement = Case(*when_statements, output_field=field)
-                if requires_casting:
-                    case_statement = Cast(case_statement, output_field=field)
-                update_kwargs[field.attname] = case_statement
-            updates.append(([obj.pk for obj in batch_objs], update_kwargs))
```

Comment:
Unfortunately performing the resolving of the expression (which is known to take a significant amount of time see ticket-31202) while the transaction is opened is likely to be more damageable than the memory tax it incurs.